### ISSUE TITLE
关于deepbdc方法的咨询

### ISSUE THREAD

**Issue by canglang007** [RL-VIG/LibFewShot#93]
最近在使用 LibFewShot 进行学习，然后关注到了 deepbdc 的方法，我采用的数据集中 train 部分一共有 200 个类别，所以我在配置文件中修改了 num_class,然后把 train_episode 设置为了 1000，但是在进行预训练的时候发现 train 部分的 epoisode 居然变成了 161，没有明白这个是什么原因。 下面是训练的日志文件的第 0 个 epoch 中 train 的情况： —————————————————————————————————————————————— 
2023-11-26 10:52:30,627 [INFO] core.trainer: ============ Train on the train set ============
2023-11-26 10:52:30,632 [INFO] core.trainer: learning rate: [0.001, 0.05]
2023-11-26 10:52:38,790 [INFO] core.trainer: Epoch-(0): [100/161]	Time 0.077 (0.081)	Calc 0.045 (0.050)	Data 0.000 (0.000)	Loss 5.149 (8.369)	Acc@1 1.562 (1.734)
2023-11-26 10:52:43,501 [INFO] core.trainer: Epoch-(0): [161/161]	Time 0.076 (0.079)	Calc 0.045 (0.048)	Data 0.000 (0.000)	Loss 5.069 (7.128)	Acc@1 0.000 (1.883)
2023-11-26 10:52:43,504 [INFO] core.trainer:  * Acc@1 1.883 
—————————————————————————————————————————————
然后是配置文件的信息：

```
augment: true
augment_method: DeepBdcAugment
augment_times: 1
augment_times_query: 1
backbone:
  kwargs:
    reduce_dim: 640
  name: resnet12Bdc
batch_size: 64
classifier:
  kwargs:
    classifier_path: null
    dropout_rate: 0.8
    emb_func_path: null
    is_distill: false
    num_class: 200
    reduce_dim: 640
    val_type: meta
  name: DeepBDC_Pretrain
data_root: ./WebCaricature
dataloader_num: 1
deterministic: true
device_ids: 0
episode_size: 1
epoch: 170
image_size: 84
includes:
- headers/data.yaml
- headers/device.yaml
- headers/misc.yaml
- headers/model.yaml
- headers/optimizer.yaml
log_interval: 100
log_level: info
log_name: null
log_paramerter: false
lr_scheduler:
  kwargs:
    gamma: 0.1
    milestones:
    - 100
    - 150
  name: MultiStepLR
n_gpu: 1
optimizer:
  kwargs:
    lr: 0.05
    momentum: 0.9
    nesterov: true
    weight_decay: 0.0005
  name: SGD
  other:
    emb_func.bdc_pool.temperature: 0.001
parallel_part:
- emb_func
port: 48518
pretrain_path: null
query_num: 15
rank: 0
result_root: ./results
resume: false
save_interval: 10
save_part:
- emb_func
- cls_classifier
seed: 2147483647
shot_num: 1
tag: null
tb_scale: 1.0
test_episode: 1000
test_epoch: 5
test_query: 15
test_shot: 1
test_way: 5
train_episode: 1000
use_memory: false
val_per_epoch: 1
warmup: 0
way_num: 5
workers: 8

```

---

希望可以得到解答

**Comment by VincenDen** [comment]
预训练不是按照episode方式训练，而是根据batchsize方式训练，具体细节需要您仔细理解一下论文或者是代码

**Comment by canglang007** [comment]
好的，非常感谢您的回答
